TWLight/resources/views.py

```diff
@@ -36,65 +34,70 @@
 
 
 class PartnersFilterView(ListView):
-    """
+    """Build the view listing all library partners.
     Since T278337, this View has passed from FilterView to ListView because we have to
     build a Partner dictionary element from the partner descriptions in a JSON file instead of
-    getting everything from the database
+    getting everything from the database.
     """
 
     model = Partner
 
     def get_queryset(self):
-        qs = (
-            Partner.objects.prefetch_related("languages")
-            .select_related("coordinator", "logos")
-            .order_by("company_name")
-        )
-        # The ordering here is useful primarily to people familiar with the
-        # English alphabet. :/
+        """Return the queryset of all partners based on user group.
+        Staff can see all resources, even those which are marked not
+        available, so we use the custom Manager even_not_available.
+        """
         if self.request.user.is_staff:
             messages.add_message(
                 self.request,
                 messages.INFO,
                 "Because you are a staff member, this page may include "
                 "Partners who are not yet available to all users.",
             )
-            qs = (
-                Partner.even_not_available.prefetch_related("languages")
-                .select_related("coordinator", "logos")
-                .order_by("company_name")
-            )
+            qs = Partner.even_not_available.prefetch_related("languages")
+        else:
+            qs = Partner.objects.prefetch_related("languages")
 
-        return qs
+        # The ordering here is useful primarily to people familiar with the
+        # English alphabet. :/
+        partner_queryset = qs.select_related("coordinator", "logos").order_by(
```

Comment:
This is a big improvement; I think you could put the prefetch here to, for even more DRYness.

Reply:
Done!